### PR TITLE
arm_auth: fix spamming while on the ground

### PR DESCRIPTION
**Pull request by NicolasM0**
**Describe the bug**
When the drone is on the ground, pre-arm checks are run continuously. If arm authorization is enabled, authorization request are also sent continuously to the authorizer and the authorization result (either accepted/reject/no response) spams mavlink text messages.

**To Reproduce**
set COM_ARM_AUTH_REQ to 1 while on the ground (no need to have an authorizer but other prearm check have to pass)

**Expected behavior**
- I suppose the arm authorization should be request only on arming and not continuously on the ground
- if arm authorization have to be checked while on the ground, it should not spam with mavlink text messages

arm_auth_spam

**Additional context**
The discussion started [here](https://github.com/PX4/PX4-Autopilot/pull/15754) but no fix have been pushed in this PR for this issue.

**Describe your solution**
Remove the arm authorization when pre-arm check are done just to update status_flags (not for arming)

**Describe possible alternatives**
Other solution have been discussed in the previous  [PR](https://github.com/PX4/PX4-Autopilot/pull/15754)

### CODE REVIEW

**Comment by NicolasM0** [comment]
@bkueng I reopened a PR to follow the discussion skipped in the previous one

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -2419,7 +2419,10 @@ Commander::run()
 			// Evaluate current prearm status
 			if (!armed.armed && !status_flags.condition_calibration_enabled) {
 				bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, status, status_flags, true, false, true, 30_s);
-				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, _safety, _arm_requirements, status, false);
+				PreFlightCheck::arm_requirements_t _arm_requirements_no_arm_auth = _arm_requirements;
```
Comment (dagar) [inline src/modules/commander/Commander.cpp:2422]:
```suggestion
				PreFlightCheck::arm_requirements_t arm_requirements_no_arm_auth = _arm_requirements;
```

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -2419,7 +2419,10 @@ Commander::run()
...
+				_arm_requirements_no_arm_auth.arm_authorization = false;
```
Comment (dagar) [inline src/modules/commander/Commander.cpp:2423]:
```suggestion
				arm_requirements_no_arm_auth.arm_authorization = false;
```

File: src/modules/commander/Commander.cpp [hunk 3]
```diff
@@ -2419,7 +2419,10 @@ Commander::run()
...
+				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, _safety, _arm_requirements_no_arm_auth,
```
Comment (dagar) [inline src/modules/commander/Commander.cpp:2424]:
```suggestion
				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, _safety, arm_requirements_no_arm_auth,
```

**Review by dagar — DISMISSED** [review]

**Comment by dagar** [comment]
There's a minor merge conflict to resolve.

File: src/modules/commander/Commander.cpp [hunk 4]
```diff
@@ -2632,7 +2632,11 @@ Commander::run()
 			// Evaluate current prearm status
 			if (!_armed.armed && !_status_flags.condition_calibration_enabled) {
 				bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, _status, _status_flags, true, false, true, 30_s);
+
+				PreFlightCheck::arm_requirements_t arm_requirements_no_arm_auth = _arm_requirements;
+				arm_requirements_no_arm_auth.arm_authorization = false;        
 				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, _arm_requirements, _status, false);
```
Comment (dagar) [inline src/modules/commander/Commander.cpp:2638]:
```suggestion
				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, arm_requirements_no_arm_auth, _status, false);
```

**Review by dagar — DISMISSED** [review]

File: src/modules/commander/Commander.cpp [hunk 5]
```diff
@@ -2632,7 +2632,11 @@ Commander::run()
 			// Evaluate current prearm status
 			if (!_armed.armed && !_status_flags.condition_calibration_enabled) {
 				bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, _status, _status_flags, true, false, true, 30_s);
-				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, _arm_requirements, _status, false);
+
+				PreFlightCheck::arm_requirements_t arm_requirements_no_arm_auth = _arm_requirements;
+				arm_requirements_no_arm_auth.arm_authorization = false;        
+				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, arm_requirements_no_arm_auth, _status, false);
```
Comment (dagar) [inline src/modules/commander/Commander.cpp:2638]:
```suggestion
				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, arm_requirements_no_arm_auth, _status, false);
				
				
				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, _status_flags, _safety, arm_requirements_no_arm_auth,
							_status, false);				
```

Comment (dagar) [reply src/modules/commander/Commander.cpp:2638]: Attempting to fix CI style failure from Github.

**Review by dagar — DISMISSED** [review]

**Pull request merged**